main/src/library/MutSet.flix

```diff
@@ -412,14 +412,14 @@ namespace MutSet {
     ///
     /// Returns an iterator over `s`.
     ///
-    pub def iterator(r1: Region[r1], s: MutSet[a, r2]): Iterator[a, r1 and r2] \ { Write(r1), Read(r2) } =
+    pub def iterator(r1: Region[r1], s: MutSet[a, r2]): Iterator[a, r1 and r2, r1] \ { Write(r1), Read(r2) } =
         let MutSet(ms) = s;
-        unsafe_cast Set.iterator(r1, deref ms) as Iterator[a, r1 and r2]
+        unsafe_cast Set.iterator(r1, deref ms) as Iterator[a, r1 and r2, r1]
```

Comment:
This looks interesting. I think the effect system is "exposing us" for the fact that iteration through the set does not really touch `r2` because the underlying set is immutable!

Does the following help get rid of the cast?

```
Set.iterator(r1, deref ms) |> Iterator.map(x -> {discard deref ms; x})
```

(Yes, I know the expression is stupid).

Reply:
If something like this can work, then I think should we apply it.

Reply:
Updated and fixed.